packages/react/src/components/cookieConsent/modal/Modal.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { VisuallyHidden } from '@react-aria/visually-hidden';
+
+import classNames from '../../../utils/classNames';
+import styles from '../CookieConsent.module.scss';
+import { CookieConsentContext, useCookieConsentUiTexts } from '../CookieConsentContext';
+import Content from '../content/Content';
+
+export function Modal(): React.ReactElement | null {
+  const cookieConsentContext = useContext(CookieConsentContext);
+  const shouldShowModal = !cookieConsentContext.hasUserHandledAllConsents();
+  const [isModalInitiallyShown] = useState<boolean>(shouldShowModal);
+  const [popupTimerComplete, setPopupTimerComplete] = useState<boolean>(false);
+  const popupDelayInMs = 500;
+  // if hasUserHandledAllConsents() was false at first and then later true, user must have saved consents.
+  const showScreenReaderSaveNotification = isModalInitiallyShown && !shouldShowModal;
+  const { settingsSaved } = useCookieConsentUiTexts();
+  useEffect(() => {
```

Comment:
This leads to Modal and everything it contains to be rendered twice. Do we need to render it twice? We probably need it for the animation? Maybe we could only render it partially based on the `popupTimerComplete` variable value? We could disable rendering of the `Details` component as it is not shown when the cookie modal opens?